nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/excel/ExcelReader.java

```diff
@@ -0,0 +1,193 @@
+                    "in the desired sheets. The schema will be derived by using those columns in the header.");
+    public static final PropertyDescriptor DESIRED_SHEETS = new PropertyDescriptor
+            .Builder().name("extract-sheets")
+            .displayName("Sheets to Extract")
+            .description("Comma separated list of Excel document sheet names whose rows should be extracted from the excel document. If this property" +
+                    " is left blank then all the rows from all the sheets will be extracted from the Excel document. The list of names is case in-sensitive. Any sheets not" +
+                    " specified in this value will be ignored. A bulletin will be generated if a specified sheet(s) are not found.")
+            .required(false)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor FIRST_ROW_NUM = new PropertyDescriptor
+            .Builder().name("excel-extract-first-row")
+            .displayName("Row number to start from")
+            .description("The row number of the first row to start processing (One based)."
+                    + " Use this to skip over rows of data at the top of a worksheet that are not part of the dataset.")
+            .required(true)
+            .defaultValue("0")
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .addValidator(StandardValidators.NON_NEGATIVE_INTEGER_VALIDATOR)
+            .build();
+
+    private AtomicReferenceArray<String> desiredSheets;
+    private volatile int firstRow;
+    private volatile String dateFormat;
+    private volatile String timeFormat;
+    private volatile String timestampFormat;
+
+    @OnEnabled
+    public void onEnabled(final ConfigurationContext context) {
+        this.firstRow = getFirstRow(context);
+        String[] rawDesiredSheets = getRawDesiredSheets(context);
+        this.desiredSheets = new AtomicReferenceArray<>(rawDesiredSheets.length);
+        IntStream.range(0, rawDesiredSheets.length)
+                .forEach(index -> this.desiredSheets.set(index, rawDesiredSheets[index]));
+        this.dateFormat = context.getProperty(DateTimeUtils.DATE_FORMAT).getValue();
+        this.timeFormat = context.getProperty(DateTimeUtils.TIME_FORMAT).getValue();
+        this.timestampFormat = context.getProperty(DateTimeUtils.TIMESTAMP_FORMAT).getValue();
+    }
+
+    private int getFirstRow(final PropertyContext context) {
+        int rawFirstRow = context.getProperty(FIRST_ROW_NUM).asInteger();
+        return getZeroBasedIndex(rawFirstRow);
+    }
+
+    static int getZeroBasedIndex(int rawFirstRow) {
+        return rawFirstRow > 0 ? rawFirstRow - 1 : 0;
+    }
+    private String[] getRawDesiredSheets(final PropertyContext context) {
```

Comment:
Methods should have a newline separating them from previous methods.

```suggestion

    private String[] getRawDesiredSheets(final PropertyContext context) {
```